Mage.Sets/src/mage/cards/l/LiberatorUrzasBattlethopter.java

```diff
@@ -0,0 +1,105 @@
+package mage.cards.l;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.common.SpellCastControllerTriggeredAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.continuous.CastAsThoughItHadFlashAllEffect;
+import mage.abilities.keyword.FlashAbility;
+import mage.abilities.keyword.FlyingAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.counters.CounterType;
+import mage.filter.FilterCard;
+import mage.filter.StaticFilters;
+import mage.filter.predicate.Predicates;
+import mage.filter.predicate.mageobject.ColorlessPredicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.game.stack.Spell;
+import mage.watchers.common.ManaPaidSourceWatcher;
+
+import java.util.UUID;
+
+/**
+ * @author PurpleCrowbar
+ */
+public final class LiberatorUrzasBattlethopter extends CardImpl {
+
+    private static final FilterCard filter = new FilterCard("colorless spells and artifact spells");
+
+    static {
+        filter.add(Predicates.or(
+                ColorlessPredicate.instance,
+                CardType.ARTIFACT.getPredicate()
+        ));
+    }
+
+    public LiberatorUrzasBattlethopter(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT, CardType.CREATURE}, "{3}");
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.THOPTER);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(2);
+
+        // Flash
+        this.addAbility(FlashAbility.getInstance());
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // You may cast colorless spells and artifact spells as though they had flash.
+        this.addAbility(new SimpleStaticAbility(
+                new CastAsThoughItHadFlashAllEffect(Duration.WhileOnBattlefield, filter)
+        ));
+
+        // Whenever you cast a spell, if the amount of mana spent to cast that spell is greater
+        // than Liberator, Urza's Battlethopter's power, put a +1/+1 counter on Liberator.
+        this.addAbility(new SpellCastControllerTriggeredAbility(
+                new LiberatorUrzasBattlethopterEffect(), StaticFilters.FILTER_SPELL_A, false, true
```

Comment:
This needs to use `ConditionalInterveningIfTriggeredAbility`.  Reason is that intervening if clauses need two checks both before it goes on the stack and before it resolves.  It shouldn't go on the stack at all if the condition isn't met.  The decorator class handles that for you.

The custom effect can be removed (`AddCountersSourceEffect` is all you need) and the logic for the condition needs to be put in a `Condition` enum.

Reply:
Thanks, fixed